64_FFT/app_resources/shader.comp.hlsl

```diff
@@ -1,54 +1,47 @@
 #include "common.hlsl"
 #include "nbl/builtin/hlsl/workgroup/fft.hlsl"
+//#include "nbl/builtin/hlsl/subgroup/fft.hlsl"
 #include "nbl/builtin/hlsl/glsl_compat/subgroup_basic.hlsl"
 
-// Users must define this function for the FFT to work
-namespace nbl
-{
-namespace hlsl
-{
-namespace glsl 
-{
-
-// Define this method from glsl_compat/core.hlsl 
-uint32_t3 gl_WorkGroupSize() {
-    return uint32_t3(_NBL_HLSL_WORKGROUP_SIZE_, 1, 1);
-}
-
-} //namespace glsl
-} //namespace hlsl
-} //namespace nbl
-
-
 [[vk::push_constant]] PushConstantData pushConstants;
 
-groupshared uint32_t sharedmem[2 * WorkgroupSize];
+groupshared input_t sharedmem[2 * WorkgroupSize];
+
+// Users MUST define this method for FFT to work
+namespace nbl { namespace hlsl { namespace glsl
+{
+uint32_t3 gl_WorkGroupSize() { return uint32_t3(WorkgroupSize, 1, 1); }
+} } }
 
 struct SharedMemoryAccessor {
-	void set(uint32_t idx, uint32_t x) {
-		sharedmem[idx] = x;
+	void set(uint32_t idx, nbl::hlsl::complex_t<input_t> x) {
+		sharedmem[idx] = x.real();
+		sharedmem[idx + WorkgroupSize] = x.imag();
 	}
 	
-	uint32_t get(uint32_t idx) {
-		return sharedmem[idx];
+	nbl::hlsl::complex_t<input_t> get(uint32_t idx) {
+		nbl::hlsl::complex_t<input_t> retVal = {sharedmem[idx], sharedmem[idx + WorkgroupSize]};
+		return retVal;
 	}
 
 	void workgroupExecutionAndMemoryBarrier() {
-        AllMemoryBarrierWithGroupSync();
+		AllMemoryBarrierWithGroupSync();
     }
 };
 
 struct Accessor {
-	void set(uint32_t idx, uint32_t x) {
-		vk::RawBufferStore<uint32_t>(pushConstants.outputAddress + sizeof(uint32_t) * idx, x);
+	void set(uint32_t idx, nbl::hlsl::complex_t<input_t> x) {
+		vk::RawBufferStore< vector<input_t, 2> >(pushConstants.outputAddress + sizeof(vector<input_t, 2>) * idx, vector<input_t, 2>(x.real(), x.imag()));
 	}
 	
-	uint32_t get(uint32_t idx) {
-		return vk::RawBufferLoad<uint32_t>(pushConstants.inputAddress + sizeof(uint32_t) * idx);
+	nbl::hlsl::complex_t<input_t> get(uint32_t idx) {
+		vector<input_t, 2> aux = vk::RawBufferLoad< vector<input_t, 2> >(pushConstants.inputAddress + sizeof(vector<input_t, 2>) * idx);
+		nbl::hlsl::complex_t<input_t> retVal = {aux.x, aux.y};
+		return retVal;
```

Comment:
btw after the DXC update you should be able to just load store `complex_t<float>`

Also why are the `input_t` and `output_t` typedefs in your code?

Reply:
btw there are some ready made stateful accessors that work with BDA
https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/include/nbl/builtin/hlsl/bda/bda_accessor.hlsl

Reply:
Haven't bothered to change much of the example so far, once I'm 100% done with workgroup I'll clean it up